staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/kcp_rest_options_getter.go

```diff
@@ -0,0 +1,33 @@
+package apiserver
+
+import (
+	"fmt"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/generic"
+)
+
+type apiBindingAwareCRDRESTOptionsGetter struct {
+	delegate generic.RESTOptionsGetter
+	crd      *apiextensionsv1.CustomResourceDefinition
+}
+
+func (t apiBindingAwareCRDRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (generic.RESTOptions, error) {
+	ret, err := t.delegate.GetRESTOptions(resource)
+	if err != nil {
+		return ret, err
+	}
+
+	if _, bound := t.crd.Annotations["apis.kcp.dev/bound-crd"]; !bound {
+		return ret, nil
+	}
+
+	apiIdentity := t.crd.Annotations["apis.kcp.dev/identity"]
+	if apiIdentity == "" {
+		return generic.RESTOptions{}, fmt.Errorf("missing 'apis.kcp.dev/identity' annotation")
+	}
+
+	ret.ResourcePrefix += "-" + apiIdentity
```

Comment:
nit: resource prefixes of CRDs cannot have a dash? I.e. there is no risk of an attack hijacking data this way by creating a normal CRD with prefix+apiIdentity as the prefix?

Reply:
Good catch. Yes they can. I tested with a resource name of `dashed-endpoints` in the APIResourceSchema is this is what is in etcd when you create one (via an APIBinding):

```
/registry/andy.io/dashed-endpoints-d9d18877572da255c263f0dfcb0e39f5a7cbf7177c68f8dec5561ec81611d3e5/root:default:andy/default/test1
```

Reply:
Looks like the plural has to be DNS1035 (lowercase alphanumeric or `-`, start with alphabetic, end with alphanumeric)

Reply:
So maybe we use `:`?

Reply:
: is fine, if that is valid (in URLs it is, in etcd too).